omisego-sdk/src/main/java/co/omisego/omisego/websocket/listener/TransactionRequestListener.kt

```diff
@@ -0,0 +1,40 @@
+package co.omisego.omisego.websocket.listener
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 23/7/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import co.omisego.omisego.model.APIError
+import co.omisego.omisego.model.transaction.consumption.TransactionConsumption
+import co.omisego.omisego.websocket.event.SocketEvent
+import co.omisego.omisego.websocket.event.TransactionConsumptionFinalizedEvent
+import co.omisego.omisego.websocket.event.TransactionConsumptionRequestEvent
+import co.omisego.omisego.websocket.strategy.FilterStrategy
+
+abstract class TransactionRequestListener(
+    final override val strategy: FilterStrategy = FilterStrategy.Event(allowedEvents)
+) : SimpleSocketCustomEventListener<SocketEvent<*>>() {
+    final override fun onSpecificEvent(event: SocketEvent<*>) {
+        when (event) {
+            is TransactionConsumptionRequestEvent -> event.socketReceive.data?.let(::onTransactionConsumptionRequest)
```

Comment:
In case the data is null, the event will just be ignored, is this ok? (shouldn't happen I guess)

Reply:
Hmm, maybe we need another function for dispatching this error.